server/plugin.go

```diff
@@ -74,37 +86,107 @@ func (p *Plugin) OnActivate() error {
 	}
 	p.BotUserID = botID
 
-	channel, appErr := p.API.GetChannelByName(team.Id, channelSplit, false)
+	p.API.LogInfo("bot user", "id", p.BotUserID)
+
+	// get or create channel and return mattermost channel id
+	err = p.getOrCreateMattermostChannels(teamChannels)
+	if err != nil {
+		return err
+	}
+
+	p.Channels = teamChannels
+
+	p.API.LogInfo("channels resvolved", "tc", teamChannels)
+	if err := p.registerCommands(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Plugin) resolveAndSetTeamIDs(channels []*TeamChannel) error {
+	//mattermostChannels := []TeamChannel{}
+	for _, teamChannel := range channels {
+		p.API.LogInfo("resolve for teamchannel", "tc", teamChannel)
+		appErr := p.resolveAndSetTeamID(teamChannel)
+		if appErr != nil {
+			return appErr
+		}
+		//mattermostChannels = append(mattermostChannels, *teamChannel)
+	}
+	return nil
+}
+
+func (p *Plugin) resolveAndSetTeamID(teamChannel *TeamChannel) error {
+	team, appErr := p.API.GetTeamByName(teamChannel.TeamName)
+	if appErr != nil {
+		return appErr
+	}
+	teamChannel.TeamID = team.Id
+	return nil
+}
+
+func parseTeamChannelsNames(teamChannel string) ([]*TeamChannel, error) {
+	channels := []*TeamChannel{}
+	splitChannels := strings.Split(teamChannel, ";")
+	for _, splitChannel := range splitChannels {
+		if len(splitChannel) < 1 {
+			continue
+		}
+		split := strings.Split(splitChannel, ",")
+		if len(split) != 2 {
+			return nil, errors.New("teamChannel setting doesn't follow the pattern $TEAM_NAME,$CHANNEL_NAME")
+		}
+		channels = append(channels, &TeamChannel{
+			TeamName:    split[0],
+			ChannelName: split[1],
+		})
+	}
+	return channels, nil
+}
+
+func (p *Plugin) getOrCreateChannel(teamChannel *TeamChannel) (string, error) {
+	p.API.LogInfo("get channel", "tc", teamChannel)
+	channel, appErr := p.API.GetChannelByName(teamChannel.TeamID, teamChannel.ChannelName, false)
 	if appErr != nil && appErr.StatusCode == http.StatusNotFound {
+		p.API.LogInfo("Channel not found")
```

Comment:
We should include the channel name in this log

Reply:
Removed the line, "Creating Channel" with channel name is logged directly after it